drivers/sensors/cirque_pinnacle.c

```diff
@@ -71,6 +74,17 @@ void cirque_pinnacle_scale_data(pinnacle_data_t* coordinates, uint16_t xResoluti
     // scale coordinates to (xResolution, yResolution) range
     coordinates->xValue = (uint16_t)(xTemp * xResolution / CIRQUE_PINNACLE_X_RANGE);
     coordinates->yValue = (uint16_t)(yTemp * yResolution / CIRQUE_PINNACLE_Y_RANGE);
+#else
+    int32_t xTemp = 0;
+    int32_t yTemp = 0;
+
+    xTemp = ((int32_t)coordinates->xDelta) * (int32_t)xResolution / (int32_t)CIRQUE_PINNACLE_X_RANGE;
+    yTemp = ((int32_t)coordinates->yDelta) * (int32_t)yResolution / (int32_t)CIRQUE_PINNACLE_Y_RANGE;
```

Comment:
Scaling deltas this way seem like it would be prone to truncation errors for small movement. I was playing around with relative mode and wasn't sure exactly how to deal with this yet; was thinking to calculate the remainder after scaling, and add the remainder to coordinates on the next set of data.

Reply:
I have not come up with a reasonable solution for now.

Reply:
What do you think of this for relative mode scaling:
```
void cirque_pinnacle_scale_data(pinnacle_data_t* coordinates, uint16_t xResolution, uint16_t yResolution) {
    int32_t        xTemp = 0, yTemp = 0;
    ldiv_t         temp;
    static int32_t xRemainder, yRemainder;

    temp       = ldiv(((int32_t)coordinates->xDelta) * (int32_t)xResolution + xRemainder, (int32_t)CIRQUE_PINNACLE_X_RANGE);
    xTemp      = temp.quot;
    xRemainder = temp.rem;

    temp       = ldiv(((int32_t)coordinates->yDelta) * (int32_t)yResolution + yRemainder, (int32_t)CIRQUE_PINNACLE_Y_RANGE);
    yTemp      = temp.quot;
    yRemainder = temp.rem;

    coordinates->xDelta = (int16_t)xTemp;
    coordinates->yDelta = (int16_t)yTemp;
}
```
Better late than never?

Reply:
Sure, I can try that

Reply:
`ldiv` is not available in QMK. I'm wondering whether this is worth it.

Reply:
Forgot `#include <stdlib.h>`.

Reply:
Added